### ISSUE TITLE
E2guardian Reporting

### ISSUE THREAD

**Issue by Dalacor** [e2guardian/e2guardian#333]
I am very happy with the content filtering of E2guardian. I think it works fast, reliably and consistently and I like the ability to control the look of the access denied page, the categories of websites blocked, phrasewords blocked etc.

However, I am finding more and more the lack of visual reports (PDF etc) as well as real time reporting is becoming a serious drawback. In addition, schools are reviewing whether to install monitoring software on staff laptops so when those laptops go home, schools can be alerted if staff access inapropriate websites at home.

I currently export the E2guardian.log file to one of my servers and use Sawmill to view the logs to see what websites are being visited and I can create reports on this. But this is problematic, because its time consuming to use Sawmill to view the logs, Sawmill is a generic log analyser so is not that useful for creating reports for teachers etc, plus its not really laid out in a format that is useful if you want to check for instances of cyberbullying etc!

In addition, the lack of real time alert reporting is a real drawback.

I like how Impero Education Pro handles Internet filtering - specifically their reporting and real time alerts, but I don't want to replace E2guardian with another product as I think the Internet Filtering of E2guardian is second to none. But its no longer sustainable to have a reporting system where I copy the log to my server where Sawmill is installed, import into Sawmill and then create very basic reports based on websites visited as this is very time consuming and the reports are not that useful for teachers. I am quite happy to continue using Sawmill if there was a way to improve reporting efficiency by setting up preset filters etc. But the realtime reporting is still a problem.

Especially with the introduction of SSL Encryption for Google now means that the Google Search Keywords no longer works as E2guardian no longer records them.

How do other people address creating useful reports for end users from E2guardian as Sawmill is not adequate for our needs and I have no realtime reporting functionality and I also need to be able to record Google Searches so that we know what kids are searching for.

Thank you.

**Comment by forid786** [comment]
What would be nice if we could have a "log" web interface that allows us to filter the blocked and access pages by user/IP/time. That would be really useful, especially in a educational setting I think. You'll never know when you may need the history of someone's browsing.

**Comment by Dalacor** [comment]
A few months back, I contacted Sawmill about this and they said that their program does support the ability to do the following:

Connect to the remote server log files via ssh etc.
It can update the sawmill log file on a schedule
It can create reports on a schedule and even email them.
If I remember correctly, I think it can create alerts as well.

I have not had time to do anything with this as you do need to spend time reviewing what you want to filter, creating preset filters to essentially customise the program to filter for what you are looking for. The biggest problem with Web Filtering is there is in a sense too much data to put into a report. Trying to decide what is useful in a report is the challenge.

Sawmill is in a web interface and it is possible to filter by user/IP/time. I need to figure out how to get pc name to show in E2guardian logs. I have set it up to use logclienthostnames = on, but it does not seem to work.

Anyway, I am going to close this question as Sawmill does have the ability to automate reporting, more or less do real time reporting and there is the ability to filter the logs quite effectively. However, it will take time to design the filters to get Sawmill to create reports that are useful.

**Comment by forid786** [comment]
Thanks to @Marcelloc I'm currently using Light Squid for reporting. Although it's not the most elegant thing ever, I can see what websites a particular IP has visited.

I'd look into Sawmill but probably our in house logging is a better bet. Although E2 Guardian is making huge progress, it's still got some way to go before it's fully polished with proper granular logging, updated phrase lists and maybe a unified Web gui that would make setting this up easier for new users.

**Comment by Dalacor** [comment]
Fair enough. Glad that you found something that works for you. I will go with Sawmill on the grounds that I can use Sawmill to provide logging for Squid, E2Guardian and my email program. It does take time to set up Sawmill in that you need to decide what to filter etc.

A web gui could make things easier, but it's not hard to setup E2guardian. The problem is lack of any real documentation.
